test/metadata.jl

```diff
@@ -0,0 +1,3065 @@
+    colmetadata!(long, :variable, "var1", "var1", style=:none)
+    colmetadata!(long, :value, "val", "val", style=:note)
+    colmetadata!(long, :value, "val1", "val1", style=:none)
+
+    res = unstack(long)
+    @test collect(metadatakeys(res)) == ["name"]
+    @test metadata(res, "name") == "empty"
+    @test isempty(colmetadatakeys(res, :id))
+    @test collect(colmetadatakeys(res, :a)) == ["name"]
+    @test colmetadata(res, :a, "name") == "a"
+    @test isempty(colmetadatakeys(res, :b))
+    @test isempty(colmetadatakeys(res, :c))
+    @test isempty(colmetadatakeys(res, :d))
+
+    res = unstack(long, :id, :variable, :value)
+    @test collect(metadatakeys(res)) == ["name"]
+    @test metadata(res, "name") == "empty"
+    @test isempty(colmetadatakeys(res, :id))
+    @test isempty(colmetadatakeys(res, :b))
+    @test isempty(colmetadatakeys(res, :c))
+    @test isempty(colmetadatakeys(res, :d))
+
+    res = unstack(long, :a, :variable, :value, valuestransform=copy)
+    @test collect(metadatakeys(res)) == ["name"]
+    @test metadata(res, "name") == "empty"
+    @test collect(colmetadatakeys(res, :a)) == ["name"]
+    @test colmetadata(res, :a, "name") == "a"
+    @test isempty(colmetadatakeys(res, :b))
+    @test isempty(colmetadatakeys(res, :c))
+    @test isempty(colmetadatakeys(res, :d))
+end
+
+@testset "permutedims" begin
+    df = DataFrame(a=["x", "y"], b=[1.0, 2.0], c=[3, 4], d=[true, false])
+    res = permutedims(df, 1)
+    @test getfield(res, :metadata) === nothing
+    @test getfield(res, :colmetadata) === nothing
+
+    metadata!(df, "name", "empty", style=:note)
+    metadata!(df, "name1", "empty1", style=:none)
+    colmetadata!(df, :a, "name", "a", style=:note)
+    colmetadata!(df, :a, "name1", "a1", style=:none)
+    colmetadata!(df, :b, "name", "b", style=:note)
+    colmetadata!(df, :b, "name1", "b1", style=:none)
+    colmetadata!(df, :c, "name", "c", style=:note)
+    colmetadata!(df, :c, "name1", "c1", style=:none)
+    colmetadata!(df, :d, "name", "d", style=:note)
+    colmetadata!(df, :d, "name1", "d1", style=:none)
+    res = permutedims(df, 1)
+    @test collect(metadatakeys(res)) == ["name"]
+    @test metadata(res, "name") == "empty"
+    @test getfield(res, :colmetadata) === nothing
+end
+
+@testset "broadcasting" begin
+    df1 = DataFrame(a=1:3, b=11:13)
+    df2 = DataFrame(a=-1:-1:-3, b=-11:-1:-13)
+
+    res = log.(df1)
+    @test getfield(res, :metadata) === nothing
+    @test getfield(res, :colmetadata) === nothing
+
+    metadata!(df1, "name", "some", style=:note)
+    metadata!(df1, "name1", "some1", style=:none)
+    colmetadata!(df1, :b, "x", "y", style=:note)
+    colmetadata!(df1, :b, "x1", "y1", style=:none)
+
+    res = log.(df1)
+    @test collect(metadatakeys(res)) == ["name"]
+    @test metadata(res, "name") == "some"
+    @test isempty(colmetadatakeys(res, :a))
+    @test collect(colmetadatakeys(res, :b)) == ["x"]
+    @test colmetadata(res, :b, "x") == "y"
```

Comment:
This is a situation where most users probably wouldn't expect column-level metadata to be preserved. But dropping it wouldn't be consistent with our general rule. I guess we'll see how it goes...

Reply:
Exactly - but I think consistency of the mental model to start with is important.